src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Core.cs

```diff
@@ -62,5 +64,69 @@ private void CheckDisposedOrStarted()
                 throw new InvalidOperationException(SR.net_http_operation_started);
             }
         }
+
+        private static async Task<Exception> ValidateAndNormalizeRequestAsync(HttpRequestMessage request)
+        {
+            bool shouldBufferContent = false;
+
+            // Add headers to define content transfer, if not present
+            if (request.HasHeaders && request.Headers.TransferEncodingChunked.GetValueOrDefault())
+            {
+                if (request.Content == null)
+                {
+                    return new HttpRequestException(SR.net_http_client_execution_error,
+                        new InvalidOperationException(SR.net_http_chunked_not_allowed_with_empty_content));
+                }
+
+                // Since the user explicitly set TransferEncodingChunked to true, we need to remove
+                // the Content-Length header if present, as sending both is invalid.
+                request.Content.Headers.ContentLength = null;
+            }
+            else if (request.Content != null && request.Content.Headers.ContentLength == null && request.Headers.TransferEncodingChunked == false)
+            {
+                // We have content, but Content-Length is not set and Transfer-Encoding was explicitly unset.
+                shouldBufferContent = true;
+            }
+            else if (request.Content != null && request.Content.Headers.ContentLength == null && request.Headers.TransferEncodingChunked == null)
+            {
+                // We have content, but Content-Length is not set and Transfer-Encoding was not specified.
+                request.Headers.TransferEncodingChunked = true;
+            }
+
+            if (request.Version.Minor == 0 && request.Version.Major == 1 && request.HasHeaders)
+            {
+                // HTTP 1.0 does not support chunking
+                if (request.Content != null && request.Headers.TransferEncodingChunked == true)
+                {
+                    shouldBufferContent = true;
+                }
+
+                // HTTP 1.0 does not support Expect: 100-continue; just disable it.
+                if (request.Headers.ExpectContinue == true)
+                {
+                    request.Headers.ExpectContinue = false;
+                }
+            }
+
+            if (shouldBufferContent)
+            {
+                var computedLength = request.Content.GetComputedOrBufferLength();
```

Comment:
`var` should not be used when the type is not obvious based on the assignment expression.

```suggestion
                long? computedLength = request.Content.GetComputedOrBufferLength();
```